lib/core/src/display/navigation/event_handler.rs

```diff
@@ -0,0 +1,326 @@
+        Ok(())
+    }
+
+    /// Initialize wheel event callback.
+    fn initialize_wheel_event
+    (self:&Rc<Self>, target:&EventTarget) -> Result<()>{
+        let ehandler = self.clone();
+        let closure = move |event:WheelEvent| {
+            let ime     = &mut ehandler.data.borrow_mut();
+            let ime     = ime as &mut InternalMouseEvent;
+            let delta_y = event.delta_y() as f32;
+            match ime {
+                InternalMouseEvent::Start(mouse_start) => {
+                    let position    = mouse_start.position;
+                    let mouse_wheel = MouseWheel { position, delta_y };
+                    *ime            = InternalMouseEvent::Wheel(mouse_wheel);
+                },
+                InternalMouseEvent::Move(mouse_move) => {
+                    let position    = mouse_move.current;
+                    let mouse_wheel = MouseWheel { position, delta_y };
+                    *ime            = InternalMouseEvent::Wheel(mouse_wheel);
+                },
+                InternalMouseEvent::Wheel(mouse_wheel) => {
+                    mouse_wheel.delta_y = delta_y;
+                },
+                InternalMouseEvent::None => ()
+            }
+        };
+        add_wheel_event(&target, closure)?.forget();
+        Ok(())
+    }
+
+    /// Checks if EventHandler has an event.
+    pub fn poll(&self) -> Event {
+        let   event : &mut InternalMouseEvent = &mut self.data.borrow_mut();
+        match event.consume() {
+            InternalMouseEvent::Start(_)         => Event::Start,
+            InternalMouseEvent::Move(mouse_move) => {
+                if let Some(start) = &mouse_move.start {
+                    match start.button {
+                        MMB => {
+                            let mut movement   =  mouse_move.current;
+                                    movement  -=  mouse_move.previous;
+                                    movement.x = -movement.x;
+                            Event::Pan(PanEvent { movement })
+                        },
+                        RMB => {
+                            let focus  = start.position;
+                            let mut amount  = mouse_move.current.y;
+                                    amount -= mouse_move.previous.y;
+                            Event::Zoom(ZoomEvent { focus, amount })
+                        },
+                        _ => Event::None
+                    }
+                } else {
+                    Event::None
+                }
+            },
+            InternalMouseEvent::Wheel(mouse_wheel) => {
+                let focus  = mouse_wheel.position;
+                let amount = mouse_wheel.delta_y;
+                Event::Zoom(ZoomEvent { focus, amount })
+            }
+            _ => Event::None
+        }
+    }
+}
+
+// =============
+// === Utils ===
+// =============
+
+/// Adds mouse event callback to target.
```

Comment:
This doc tells that this funtion ads something to a target, but this function returns some dyn fn - what is this? Docs does not mention it.

Reply:
Is it cleaner?
```rust
/// Adds a generated wasm_bindgen::prelude::Closure from the `closure` param
/// to the `EventTarget` and returns it.
```